Makefile

```diff
@@ -58,7 +58,7 @@ $(TARGET_BIN):
 
 binaries: $(TARGET_BIN)
 
-install:
+install: binaries
 	$(INSTALL) -D -m 755 $(CURDIR)/bin/$(TARGET_BIN) $(BINDIR)/$(TARGET_BIN)
```

Comment:
This will break `make && sudo make install && make clean`, as `make clean` will be unable to remove the binary that is built during `sudo make install`

Reply:
Sorry, I didn't get you here. It's working fine for me.
```
$ make && make install && make clean
GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-s -w -X github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc/version.Version=v1.0.4-23-gd007ead -X github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc/version.Revision=d007ead4d04d51423aca5d76c7b30cf25f890a1d" -o bin/containerd-fuse-overlayfs-grpc github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc
GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-s -w -X github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc/version.Version=v1.0.4-23-gd007ead -X github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc/version.Revision=d007ead4d04d51423aca5d76c7b30cf25f890a1d" -o /root/fuse-overlayfs-snapshotter/bin/containerd-fuse-overlayfs-grpc github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc
install -D -m 755 /root/fuse-overlayfs-snapshotter/bin/containerd-fuse-overlayfs-grpc /usr/local/bin/containerd-fuse-overlayfs-grpc
rm -f -r /root/fuse-overlayfs-snapshotter/bin /root/fuse-overlayfs-snapshotter/_output
```

Reply:
That is working because you are running everything as root

Reply:
I checked in a fedora host.
```
➜ $ ls -ld /usr/
drwxr-xr-x. 1 root root 100 Aug 30 10:48 /usr/
[Tue 08.Nov.2022 17:22:31] sshedi@ ~ ➤
➜ $ ls -ld /usr/local
drwxr-xr-x. 1 root root 90 Aug  9 19:27 /usr/local
[Tue 08.Nov.2022 17:22:33] sshedi@ ~ ➤
➜ $ ls -ld /usr/local/bin
drwxr-xr-x. 1 root root 228 Oct 25 13:19 /usr/local/bin
```

So it's evident that regular user can't write to `/usr/local/bin`, using `sudo` is a must.

But `make clean` works fine as regular user, no problems.
```
test@ph4dev [ ~/fuse-overlayfs-snapshotter ]$ ls
bin       cmd         fuseoverlayfs.go       go.mod  LICENSE      Makefile  README.md
check.go  Dockerfile  fuseoverlayfs_test.go  go.sum  MAINTAINERS  plugin

test@ph4dev [ ~/fuse-overlayfs-snapshotter ]$ ls bin
containerd-fuse-overlayfs-grpc

test@ph4dev [ ~/fuse-overlayfs-snapshotter ]$ make clean
rm -f -r /home/test/fuse-overlayfs-snapshotter/bin /home/test/fuse-overlayfs-snapshotter/_output
```

So, I didn't get your expectation here, can you please explain more?

Reply:
`make && sudo make install && make clean' seems to work, but `sudo make install && make clean` doesn't work for me

```console
$ make clean
rm -f -r /home/suda/gopath/src/github.com/containerd/fuse-overlayfs-snapshotter/bin /home/suda/gopath/src/github.com/containerd/fuse-overlayfs-snapshotter/_output
rm: cannot remove '/home/suda/gopath/src/github.com/containerd/fuse-overlayfs-snapshotter/bin/containerd-fuse-overlayfs-grpc': Permission denied
make: *** [Makefile:68: clean] Error 1
```

Reply:
Got the problem.
`make && sudo make installl && make clean` this will create binaries as regular user and install as sudo and make clean is no problem.

But with `sudo make install && make clean` (no make at first as regular user), binaries get created as sudo hence regular user can't delete it.